packages/calypso-e2e/src/test/rest-api-client.test.ts

```diff
@@ -0,0 +1,179 @@
+import { describe, expect, test, jest } from '@jest/globals';
+import nock from 'nock';
+import { RestAPIClient, BEARER_TOKEN_URL } from '../rest-api-client';
+import { SecretsManager } from '../secrets';
+import type { Secrets } from '../secrets';
+
+const fakeSecrets = {
+	calypsoOauthApplication: {
+		client_id: 'some_value',
+		client_secret: 'some_value',
+	},
+	testAccounts: {
+		basicUser: {
+			username: 'wpcomuser2',
+			password: 'hunter2',
+			primarySite: 'wpcomuser.wordpress.com/',
+		},
+		noUrlUser: {
+			username: 'nourluser',
+			password: 'password1234',
+		},
+	},
+} as unknown as Secrets;
+
+jest.spyOn( SecretsManager, 'secrets', 'get' ).mockImplementation( () => fakeSecrets );
+
+describe( 'RestAPIClient: getBearerToken', function () {
+	test( 'Bearer Token is returned upon successful request', async function () {
+		const mockedToken = 'abcdefghijklmn';
+
+		nock( BEARER_TOKEN_URL )
+			.post( /.*/ )
+			.reply( 200, {
+				success: true,
+				data: {
+					bearer_token: mockedToken,
+					token_links: [ 'link_1', 'link_2' ],
+				},
+			} );
+
+		const restAPIClient = new RestAPIClient( {
+			username: 'user',
+			password: 'password',
+		} );
+		const bearerToken = await restAPIClient.getBearerToken();
+
+		expect( bearerToken ).toBeDefined();
+		expect( bearerToken ).toBe( mockedToken );
+		expect( typeof bearerToken ).toBe( 'string' );
```

Comment:
(optional small nit) -- I think you might only need the `toBe` check here. AFAIK, it's a referential comparison, so if `bearerToken` is not a string or is undefined, that check should fail!

Reply:
It doesn't look like this is the case:

```
    Expected: "string"
    Received: "abcdefghijklmn"

      47 | 		expect( bearerToken ).toBeDefined();
      48 | 		expect( bearerToken ).toBe( mockedToken );
    > 49 | 		expect( bearerToken ).toBe( 'string' );
```